edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Service/Program.cs

```diff
@@ -320,19 +322,19 @@ static string GetLocalConfigFilePath(IConfiguration configuration, ILogger logge
             return localConfigPath;
         }
 
-        static Option<ulong> GetStorageMaxTotalWalSizeIfExists(IConfiguration configuration)
+        static Option<T> GetStorageConfigIfExists<T>(string fieldName, IConfiguration configuration)
         {
-            ulong storageMaxTotalWalSize = 0;
+            T storageParamValue = default(T);
             try
             {
-                storageMaxTotalWalSize = configuration.GetValue<ulong>(Constants.StorageMaxTotalWalSize);
+                storageParamValue = configuration.GetValue<T>(fieldName);
             }
             catch
             {
                 // ignored
             }
 
-            return storageMaxTotalWalSize <= 0 ? Option.None<ulong>() : Option.Some(storageMaxTotalWalSize);
+            return EqualityComparer<T>.Default.Equals(storageParamValue, default(T)) ? Option.None<T>() : Option.Some(storageParamValue);
```

Comment:
please test if you provide a negative value the conguration.GetValue throws so this method will return Option.None<T>

Reply:
I tested with value (-1) and the LOG is using `NONE` config.